### PR TITLE
[DDW-917] Add block creation table to the previous epoch page

### PR DESCRIPTION
**Pull request by DeeJayElly**
This PR implemented pagination component.
[YouTrack Ticket](https://iohk.myjetbrains.com/youtrack/issue/DDW-917)

## Todos
- [x] Implement blocks creation component for previous epoch page

## Screenshots
Screenshot 2019-10-21 22 54 59

Screenshot 2019-10-21 22 59 57

## Testing Checklist
- [Slack QA thread](https://input-output-rnd.slack.com/archives/GGKFXSKC6/p1571691400194700)
- [x] Go to `Pages` => `Epoch Page` story on storybook and check if block creation page is in epoch page

### Basics

- [ ] PR has been assigned and has appropriate labels (`feature`/`bug`/`chore`, `release-x.x.x`)
- [ ] PR is updated to the most recent version of the target branch (and there are no conflicts)
- [ ] PR has a good description that summarizes all changes and shows some screenshots or animated GIFs of important UI changes
- [ ] CHANGELOG entry has been added to the top of the appropriate section (*Features*, *Fixes*, *Chores*) and is linked to the correct PR on GitHub
- [ ] Automated tests: All acceptance and unit tests are passing (`yarn test`)
- [ ] Manual tests (minimum tests should cover newly added feature/fix): App works correctly in *development* build (`yarn dev`)
- [ ] Manual tests (minimum tests should cover newly added feature/fix): App works correctly in *production* build (`yarn package` / CI builds)
- [ ] There are no *flow* errors or warnings (`yarn flow:test`)
- [ ] There are no *lint* errors or warnings (`yarn lint`)
- [ ] There are no *prettier* errors or warnings (`yarn prettier:check`)
- [ ] There are no missing translations (running `yarn manage:translations` produces no changes)
- [ ] Text changes are proofread and approved (Jane Wild / Amy Reeve)
- [ ] Japanese text changes are proofread and approved (Junko Oda)
- [ ] UI changes look good in all themes (Alexander Rukin)
- [ ] Storybook works and no stories are broken (`yarn storybook`)
- [ ] In case of dependency changes `yarn.lock` file is updated

### Code Quality
- [ ] Important parts of the code are properly commented and documented
- [ ] Code is properly typed with flow
- [ ] React components are split-up enough to avoid unnecessary re-renderings
- [ ] Any code that only works in main process is neatly separated from components

### Testing
- [ ] New feature/change is covered by acceptance tests
- [ ] New feature/change is manually tested and approved by QA team
- [ ] All existing acceptance tests are still up-to-date
- [ ] New feature/change is covered by Daedalus Testing scenario
- [ ] All existing Daedalus Testing scenarios are still up-to-date

### After Review
- [ ] Merge the PR
- [ ] Delete the source branch
- [ ] Move the ticket to `done` column on the YouTrack board
- [ ] Update Slack QA thread by marking it with a green checkmark

### CODE REVIEW

**Comment by a-rukin** [comment]
image

How about using left-padding/margin for the columns? So we can avoid situations when on resize elements are sticked to each other 

image

plus can we add elipsis so the stake pool line would be single?

**Comment by a-rukin** [comment]
there are some bugs in pages for this pr 

image

and

image

padding after this block is a bit bigger

image

sticked too mush

File: source/features/blocks/components/BlockCreationRow.tsx [hunk 1]
```diff
@@ -0,0 +1,55 @@
+import React, { Component } from 'react';
```
Comment (yakovkaravelov) [inline source/features/blocks/components/BlockCreationRow.tsx]: @DeeJayElly Same here. Would you please use functional component instead which is better in terms of performance?

File: source/features/blocks/components/BlockListRow.tsx [hunk 2]
```diff
@@ -1,5 +1,5 @@
-import { observer } from 'mobx-react-lite';
 import moment from 'moment';
+import React, { Component } from 'react';
```
Comment (yakovkaravelov) [inline source/features/blocks/components/BlockListRow.tsx]: @DeeJayElly What about using `useState` hook with functional component?

File: source/features/widgets/circular-progress/components/CircularProgress.tsx [hunk 3]
```diff
@@ -25,11 +25,11 @@ const CircularProgress = (props: ICircularProgressProps) => {
   let percentageCircleStyle = null;
   if (percentage <= 50) {
     percentageCircleStyle = {
-      backgroundImage: `linear-gradient(238deg, transparent 50%, #404250 50%), linear-gradient(${rotateDegree}deg, #1fc1c3 50%, #404250 50%)`,
+      backgroundImage: `linear-gradient(238deg, transparent 50%, #404250 50%), linear-gradient(${rotateDegree}deg, #eb2256 50%, #404250 50%)`,
```
Comment (yakovkaravelov) [inline source/features/widgets/circular-progress/components/CircularProgress.tsx]: @DeeJayElly I made this component using static color before themes were setup.
Now we have themes setup, style variables are there.
So we need to use color style variables here to apply styles to this `Circular Progress` component.
Seems like we should have style variables in js code as well as scss to use them in such case.
You can implement it here or if it takes a bit long, we can create another ticket or postpone that to next round.

File: source/pages/transaction.tsx [hunk 4]
```diff
@@ -1,26 +1,101 @@
...
-    .default;
-
   TransactionPage = () => (
     <NoSSR>
       <Layout>
-        <HeaderContainer />
-        <FooterContainer />
+        <div className={styles.transactionLayout}>
+          <Header withSearch brandType={BrandType.SHRINKED} />
+          <div className={styles.container}>
```
Comment (yakovkaravelov) [inline source/pages/transaction.tsx]: Same as previous PR. We can reuse `Container` component.

**Comment by IuliaDolishniak** [comment]
screen
too much space between columns

**Comment by IuliaDolishniak** [comment]
screen
too close

File: source/features/blocks/components/BlockCreationRow.tsx (hunk 1 above)
Comment (DeeJayElly) [reply source/features/blocks/components/BlockCreationRow.tsx]: Fixed

File: source/features/blocks/components/BlockListRow.tsx (hunk 2 above)
Comment (DeeJayElly) [reply source/features/blocks/components/BlockListRow.tsx]: Fixed

File: source/pages/transaction.tsx (hunk 4 above)
Comment (DeeJayElly) [reply source/pages/transaction.tsx]: Fixed

**Comment by a-rukin** [comment]
@DeeJayElly as these tickers can't guide us anywhere let's make them just bold, instead of link 
updated https://zpl.io/2G1GNZJ and https://zpl.io/2E1RrQ8

image

plus 

image

can we remove these rectangles? (they doesn't make sense for now)

**Comment by yakovkaravelov** [comment]
@DeeJayElly I've just approved for my comments. You can merge once Alexander Rukin's comment is finished.

**Comment by DeeJayElly** [comment]
> @DeeJayElly as these tickers can't guide us anywhere let's make them just bold, instead of link
> updated https://zpl.io/2G1GNZJ and https://zpl.io/2E1RrQ8
> 
> image
> 
> plus
> 
> image
> 
> can we remove these rectangles? (they doesn't make sense for now)

Fixed

File: source/features/widgets/circular-progress/components/CircularProgress.tsx (hunk 3 above)
Comment (DeeJayElly) [reply source/features/widgets/circular-progress/components/CircularProgress.tsx]: Lets postpone this to another ticket.

**Comment by DeeJayElly** [comment]
> image
> 
> How about using left-padding/margin for the columns? So we can avoid situations when on resize elements are sticked to each other
> 
> image
> 
> plus can we add elipsis so the stake pool line would be single?

Fixed

**Comment by DeeJayElly** [comment]
> screen
> too close

Fixed

**Comment by DeeJayElly** [comment]
> screen
> too much space between columns

Fixed

**Comment by DeeJayElly** [comment]
> there are some bugs in pages for this pr:
> 
> padding after this block is a bit bigger
> 
> image
> 
> image
> 
> and
> 
> image
> 
> sticked too mush

Fixed

**Comment by DeeJayElly** [comment]
> screen
> too much space between columns

Fixed

**Comment by IuliaDolishniak** [comment]
screen
@a-rukin please advise if the button should be right aligned here and if we should do something with all these space between and after the last column?

**Comment by DeeJayElly** [comment]
> screen
> @a-rukin please advise if the button should be right aligned here and if we should do something with all these space between and after the last column?

Which resolution is this? Some smaller screen width?

**Comment by a-rukin** [comment]
> screen
> @a-rukin please advise if the button should be right aligned here and if we should do something with all these space between and after the last column?

button should be sticked to the right, yes

regarding free space: can you tell a bit more how you achieved this? May be this is some edge case when some columns are droped

**Comment by a-rukin** [comment]
> image

regarding this one I also meant that the "address" line is sticked to the search bar

**Comment by IuliaDolishniak** [comment]
@DeeJayElly @a-rukin hope it helps
proof

**Comment by DeeJayElly** [comment]
> @DeeJayElly @a-rukin hope it helps
> proof

Fixed it.

**Comment by IuliaDolishniak** [comment]
Button is still left aligned
screen

**Comment by DeeJayElly** [comment]
> Button is still left aligned
> screen

Forgot to push all changes.

Pls recheck it now, after the new build

**Comment by DeeJayElly** [comment]
> > image
> 
> regarding this one I also meant that the "address" line is sticked to the search bar

Fixed

**Comment by IuliaDolishniak** [comment]
screen
@DeeJayElly can we do anything with the free space after the column? may be we can leave one column to fill this space?

**Comment by a-rukin** [comment]
@DeeJayElly mb use % for this with min-width?

**Comment by DeeJayElly** [comment]
> screen
> @DeeJayElly can we do anything with the free space after the column? may be we can leave one column to fill this space?

@IuliaDolishniak i added one more column here

**Comment by yakovkaravelov** [comment]
@DeeJayElly how can this PR has 104 commits?
Did you create this PR based off very old branch?
Would you please make sure to create new branch based off most recent develop when starting new feature to avoid such cases?
If these 104 commits are all feedback/correction ones, then it's ok, but I am not sure if such thing can happen in reality.

**Pull request merged**